pkgs/tools/filesystems/bcachefs-tools/default.nix

```diff
@@ -19,21 +20,38 @@
 , rustPlatform
 , makeWrapper
 , writeScript
+, python3
 , fuseSupport ? false
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "bcachefs-tools";
-  version = "1.3.5";
-
+  version = "1.4.0";
 
   src = fetchFromGitHub {
     owner = "koverstreet";
     repo = "bcachefs-tools";
     rev = "v${finalAttrs.version}";
-    hash = "sha256-Yq631LPpWal0hsEJS0dOtiox1295tYgUWJVIw+bsbnw=";
+    hash = "sha256-zb2ROcOZ9e4/h9OM/iSEO9VfgIYzLnZSBHVVi/ipLNM=";
   };
 
+  # FIXME: Remove next versioned tag
+  patches = [
+    (fetchpatch {
+      name = "use-libexec-instead-of-lib.patch";
+      url = "https://github.com/koverstreet/bcachefs-tools/commit/89abdd87271e237141a9d4f44d531f7c53353b83.patch";
+      hash = "sha256-rVuHoYeYb+aKH09yNERKdKpW+S+3GYW9HYm3MrzJG3c=";
+    })
+  ];
+
+  # FIXME: Is this a bug or a nix-ism?
+  # Tries to install to the 'systemd-minimal' and 'udev' nix installation paths
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace "\$(shell \$(PKG_CONFIG) --variable=systemdsystemunitdir systemd)" "$out/lib/systemd/system" \
```

Comment:
Make allows usually variables to be overridden like this:

```
installFlags = [ 
  "PKGCONFIG_SERVICEDIR=$(out)/lib/systemd/system" 
  "PKGCONFIG_UDEVDIR=$(out)/lib/udev"
];
```

This should be more robust.
Otherwise I would say that usually programs should stick to there configured installation prefix and leave it to the distributor to make sure it ends up in the right place. This also allows easier testing without requiring root in general.

Reply:
Thanks for pointing this out!